R/mod_body_coc_competition.R

```diff
@@ -0,0 +1,475 @@
+            "Exits to Permanent Housing" = ExitsToPHDQ,
+            "Benefits & Health Insurance at Exit" = BenefitsAtExitDQ,
+            "Average Length of Stay" = AverageLoSDQ,
+            "Living Situation at Entry" = LHResPriorDQ,
+            "No Income at Entry" = NoIncomeAtEntryDQ,
+            "Median Homeless History Index" = MedianHHIDQ,
+            "Long Term Homeless" = LTHomelessDQ,
+            "VISPDAT Completion at Entry" = ScoredAtEntryDQ,
+            "Housing First" = HousingFirstDQ,
+            "Prioritization of Chronic" = ChronicPrioritizationDQ
+          ) |>
+          tidyr::pivot_longer(cols = everything(),
+                       names_to = "Measure",
+                       values_to = "DQflag")
+        
+        possible_score <- pe_summary_final_scoring |>
+          dplyr::filter(AltProjectName == input$pe_provider) |>
+          dplyr::select(
+            "Exits to Permanent Housing" = ExitsToPHPossible,
+            "Benefits & Health Insurance at Exit" = BenefitsAtExitPossible,
+            "Average Length of Stay" = AverageLoSPossible,
+            "Living Situation at Entry" = LHResPriorPossible,
+            "No Income at Entry" = NoIncomeAtEntryPossible,
+            "Median Homeless History Index" = MedianHHIPossible,
+            "Long Term Homeless" = LongTermHomelessPossible,
+            "VISPDAT Completion at Entry" =
+              ScoredAtEntryPossible,
+            "Data Quality" = DQPossible,
+            "Prioritization Workgroup" = PrioritizationWorkgroupPossible,
+            "Housing First" = HousingFirstPossible,
+            "Prioritization of Chronic" = ChronicPrioritizationPossible
+          ) |>
+          tidyr::pivot_longer(cols = everything(),
+                       names_to = "Measure",
+                       values_to = "Possible Score")
+        
+        calculation <- pe_summary_final_scoring |>
+          dplyr::filter(AltProjectName == input$pe_provider) |>
+          dplyr::select(
+            "Exits to Permanent Housing" = ExitsToPHMath,
+            "Benefits & Health Insurance at Exit" = BenefitsAtExitMath,
+            "Average Length of Stay" = AverageLoSMath,
+            "Living Situation at Entry" = LHResPriorMath,
+            "No Income at Entry" = NoIncomeAtEntryMath,
+            "Median Homeless History Index" = MedianHHIMath,
+            "Long Term Homeless" = LongTermHomelessMath,
+            "VISPDAT Completion at Entry" =
+              ScoredAtEntryMath,
+            "Data Quality" = DQMath,
+            # "Prioritization Workgroup" = PrioritizationWorkgroupMath,
+            # "Housing First" = HousingFirstMath,
+            # "Prioritization of Chronic" = ChronicPrioritizationMath
+          ) |>
+          tidyr::pivot_longer(cols = everything(),
+                       names_to = "Measure",
+                       values_to = "Calculation")
+        
+        psh <- estimated_score |> dplyr::left_join(dq, by = "Measure") |>
```

Comment:
These joins are repetitive as well, they should be extracted out of this and the following code chunks and assigned to a single intermediate variable that is used in subsequent chunks

Reply:
The same is true for the `mutate` statement. That should be a part of your intermediate variable chunk